### PR TITLE
feat: add `CORRELATION` UDAF

### PR DESCRIPTION
**Pull request by reneesoika**
### Description 
Follow-up to #9361. Adds a new UDAF, `CORRELATION`, to compute the Pearson correlation coefficient between two `DOUBLE` columns in a stream or table.

### Testing done 
Added unit tests and QTTs for the new UDAF.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by JimGalasyn — APPROVED** [review]
Cool, LGTM!

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/correlation-udaf.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        {"topic": "OUTPUT", "key": "bob", "value": {"CORRELATION": "NaN"}},
+        {"topic": "OUTPUT", "key": "alice", "value": {"CORRELATION": 1.0}},
+        {"topic": "OUTPUT", "key": "alice", "value": {"CORRELATION": 0.7580764521133173}},
+        {"topic": "OUTPUT", "key": "bob", "value": {"CORRELATION": "NaN"}},
+        {"topic": "OUTPUT", "key": "bob", "value": {"CORRELATION": 1.0}},
+        {"topic": "OUTPUT", "key": "alice", "value": {"CORRELATION": 0.7580764521133173}}
+      ]
+    },
+    {
+      "name": "correlation - DELIMITED",
```
[the remark this answers is not in the document]
Comment (reneesoika) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/correlation-udaf.json:32]: The exact same test is in [average-udaf.json](https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/query-validation-tests/average-udaf.json). I don't think it works with DELIMITED, unless I am missing something?

[the remark this answers is not in the document]
Comment (jzaralim) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/correlation-udaf.json:32]: Ah my bad, I was looking at https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/query-validation-tests/average.json, which i now realize does not test average at all.

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/correlation/CorrelationUdaf.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import io.confluent.ksql.util.Pair;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+@UdafDescription(name = "correlation",
+        description = "Computes the Pearson correlation coefficient between two columns of data.",
+        author = KsqlConstants.CONFLUENT_AUTHOR
+)
+public class CorrelationUdaf implements TableUdaf<Pair<Double, Double>, Struct, Double> {
```
[the remark this answers is not in the document]
Comment (jnh5y) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/correlation/CorrelationUdaf.java:30]: Hmm....  could we support decimals as well?  Or does that get into a world of hurt?

[the remark this answers is not in the document]
Comment (reneesoika) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/correlation/CorrelationUdaf.java:30]: Supporting decimals wouldn't be bad except I don't see an existing square root method for `BigDecimal`. I'm not sure how involved that would be or if there is a library that would have that.

**Review by jzaralim — APPROVED** [review]

**Pull request merged**